### PR TITLE
feat: make lifi swapper wallet agnostic

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Upgrades lifi swapper to the wallet-agnostic pattern as part of our move to an SDK style integration of swappers in web.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

progresses https://github.com/shapeshift/web/issues/5024 (but doesn't close it)

## Risk

High risk of broken swappers, especially lifi. Specifically trade execution.

## Testing

* Check all swappers still work
* Check lifi trades still work

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)
image
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5223** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5223" target="_blank">Graphite</a>  👈
    * **PR #5224** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5224" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5223?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
Happy with this code-wise, I'll defer to @MBMaria for testing trades still work as expected.

File: src/lib/swapper/swappers/LifiSwapper/LifiSwapper.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const lifiSwapper: Swapper = {
     return Promise.resolve(
       [
         ...filterCrossChainEvmBuyAssetsBySellAssetId(input),
-        // TODO(gomes): This is weird but a temporary product compromise to accomodate for the fact that OP rewards have weird heuristics
+        // TODO(gomes): This is weird but a temporary product compromise to accommodate for the fact that OP rewards have weird heuristics
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/LifiSwapper/LifiSwapper.ts:24]: Excuse my baguette, I cannot spell this word to save me life

**Review by gomesalexandre — APPROVED** [review]
Conceptual stamp as this PR and its #5224 and #5225 frens are being tested by ops in a dedicated build currently

**Pull request merged**